IdentityServer/Core/ResourceOwnerPasswordValidator.cs

```diff
@@ -0,0 +1,59 @@
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using IdentityModel;
+using IdentityServer.Core.Users;
+using IdentityServer4.Models;
+using IdentityServer4.Validation;
+
+namespace IdentityServer.Core
+{
+    public class ResourceOwnerPasswordValidator : IResourceOwnerPasswordValidator
+    {
+        private readonly IUserIdentityService _userIdentityService;
+
+
+        public ResourceOwnerPasswordValidator(IUserIdentityService userIdentityService)
+        {
+            _userIdentityService = userIdentityService;
+        }
+
+        public async Task ValidateAsync(ResourceOwnerPasswordValidationContext context)
+        {
+            try
+            {
+                var userName = context.UserName;
```

Comment:
For me this variable is useless (logic is not that complicated and object property is well documented). Also you are here inconsistent because later you use directly context.Password 😇

Reply:
Yep.